pubsub/pulsar/pulsar.go

```diff
@@ -22,7 +22,7 @@ import (
 	"time"
 
 	"github.com/apache/pulsar-client-go/pulsar"
-	lru "github.com/hashicorp/golang-lru"
+	lru "github.com/hashicorp/golang-lru/v2"
```

Comment:
Reviewers: this file contains some actual (small) code changes since the v2 of the library uses generics